stacbuilder/pathparsers.py

```diff
@@ -173,6 +173,62 @@ def type_converters(self) -> TypeConverterMapping:
         return self._type_converters
 
 
+class GeneralInputPathParser(RegexInputPathParser):
+    """Path parser that uses regular expressions to extract properties from a path.
+    This input parser overrides the datetime, start_datetime and end_datetime properties.
+    """
+
+    def __init__(self, period: Optional[str] = None, *args, **kwargs) -> None:
+        type_converters = {
+            "year": int,
+            "month": int,
+            "day": int,
+        }
+        self._set_period(period)
+        super().__init__(type_converters=type_converters, *args, **kwargs)
+
+    def _set_period(self, period: Optional[str] = None):
+        period = period or "Daily"
```

Comment:
For parameters that select a choice from a set of fixed values we should consider an Enum or a StrEnum.

If you want to keep the string values, then make them all lower case. Capitalization is a bit uncommon for such values (usually either all lower case or all uppercase) and it might make the code more error prone.